src/html-error.c

```diff
@@ -172,21 +172,33 @@ int send_http_error_message (struct conn_s *connptr)
             "<p><em>Generated by %s.</em></p>\n" "</body>\n"
             "</html>\n";
 
-        const char p_auth_str[] =
-            "Proxy-Authenticate: Basic realm=\""
-            PACKAGE_NAME "\"\r\n";
-
-        const char w_auth_str[] =
-            "WWW-Authenticate: Basic realm=\""
-            PACKAGE_NAME "\"\r\n";
-
 	/* according to rfc7235, the 407 error must be accompanied by
            a Proxy-Authenticate header field. */
-        const char *add = connptr->error_number == 407 ? p_auth_str :
-                          (connptr->error_number == 401 ? w_auth_str : "");
+        const char *auth_str_type = 
+                connptr->error_number == 407 ? "Proxy-Authenticate" :
+                (connptr->error_number == 401 ? "WWW-Authenticate" : "");
+
+        const char auth_str_tpl[] = "%s: Basic realm=\"%s\"\r\n";
+        char* auth_str_add = NULL;
+
+        if (auth_str_type[0] != 0) {
+                int auth_str_size = snprintf(NULL, 0, auth_str_tpl, 
+                                        auth_str_type, config->basicauth_realm) + 1;
+                if (auth_str_size > 0) {
+                        auth_str_add = safemalloc(auth_str_size);
+                        if (auth_str_add != NULL) {
+                                snprintf(auth_str_add, auth_str_size, auth_str_tpl,
+                                        auth_str_type, config->basicauth_realm);
+                        }
+                }
+        }
 
         send_http_headers (connptr, connptr->error_number,
-                           connptr->error_string, add);
+                           connptr->error_string, auth_str_add ? auth_str_add : "");
+
+        if (auth_str_add != NULL) {
```

Comment:
for brevity, let's just make this `if (auth_str_add) safefree (auth_str_add);`

Reply:
also, the convention used in the code is to always add a space when calling a function between function name and opening parens. not a big deal but we should stick for it.

Reply:
I see. Sorry i had not recognized this convention.

I tried to use "tinyproxy-indent.sh", which is in this repo.
But it destroyed completely the files / added a very big amount of other changes ... so i simply had given up to try to use it.

Reply:
@rofl0r 
I have now also considered these hints in addition.
Thank for open transparent input from you ... very helpfull

Please have again a look